engine/include/cubos/engine/input/input.hpp

```diff
@@ -0,0 +1,80 @@
+/// @file
+/// @brief Contains the Input class.
+#pragma once
+
+#include <cubos/core/io/keyboard.hpp>
+#include <cubos/core/io/window.hpp>
+
+#include <cubos/engine/input/bindings.hpp>
+
+namespace cubos::engine
+{
+    /// Used to store the input bindings for multiple players and update its state accordingly as events are received.
+    class Input final
+    {
+    public:
+        Input() = default;
+        ~Input() = default;
+
+        /// Clears all bindings
+        void clear();
+
+        /// Clears all bindings for a specific player
+        /// @param player The player whose bindings will be cleared
+        void clear(int player);
+
+        /// Sets the bindings for a specific player
+        /// @param bindings The bindings to set
+        /// @param player The player whose bindings will be set
+        void bind(const InputBindings& bindings, int player = 0);
+
+        /// Gets an action state for a specific player
+        /// @param actionName The name of the action
+        /// @param player The player whose action state will be retrieved
+        /// @return True if the action exists and is pressed, false otherwise.
+        bool pressed(const std::string& actionName, int player = 0) const;
```

Comment:
Prefer using `std::string_view` over `const std::string&`

Reply:
had to use `const char*` because of the unordered_map find()